src/main/java/services/ParseSection.java

```diff
@@ -99,4 +107,59 @@ public static SectionAttribute parse(@NotNull String rawData) {
         secData.getOrDefault("Notes", "See Description. None otherwise"),
         secData.get("Room"));
   }
+
+
+  public static void parseBuilding(Map<String, String> secData, String link) {
+    List<String> lines = Utils.asResourceLines("/building.txt");
+    Map<String, String> buildings = new HashMap<>();
+    lines.stream().map(str -> str.split(",",2)).forEach(strings -> {
+      buildings.put(strings[0], strings[1]);
+    });
+
+    String location = secData.get("Room");
+    String room = "";
+    String building = null;
+
+    if(location.contains("Loc") || location.contains("Loc:")) {
+      location = location.split("Loc")[0];
+      if(Pattern.compile( "[0-9]" ).matcher(location).find()) {
```

Comment:
Same here, can you move the compiled pattern to a static so we can reuse it every run?